Framework/Algorithms/src/DiscusMultipleScatteringCorrection.cpp

```diff
@@ -172,65 +178,97 @@ std::map<std::string, std::string> DiscusMultipleScatteringCorrection::validateI
     return issues;
   }
   Geometry::IComponent_const_sptr sample = inputWS->getInstrument()->getSample();
-  if (!sample) {
+  if (!sample)
     issues["InputWorkspace"] = "Input workspace does not have a Sample";
-  } else {
-    if (inputWS->sample().hasEnvironment())
-      issues["InputWorkspace"] = "Sample must not have a sample environment";
 
+  bool atLeastOneValidShape = inputWS->sample().getShape().hasValidShape();
+  if (!atLeastOneValidShape) {
+    if (inputWS->sample().hasEnvironment()) {
+      auto env = &inputWS->sample().getEnvironment();
+      for (size_t i = 0; i < env->nelements(); i++) {
+        if (env->getComponent(i).hasValidShape()) {
+          atLeastOneValidShape = true;
+          break;
+        }
+      }
+    }
+  }
+  if (!atLeastOneValidShape) {
+    issues["InputWorkspace"] = "Either the Sample or one of the environment parts must have a valid shape.";
+  }
+
+  size_t nEnvComponents = 0;
+  if (inputWS->sample().hasEnvironment())
+    nEnvComponents = inputWS->sample().getEnvironment().nelements();
+  if (inputWS->sample().getShape().hasValidShape())
```

Comment:
I can't reuse atLeastOneValidShape here because I need to know the exact number of valid shapes. If there are 3 valid shapes for example I need to step through the 3 shapes and check each one has a valid material set up.
Having said that I will make a small change to tidy the code up here. I should be calling hasValidShape on all the possible shapes that are present whereas I only currently do it on the sample shape. I need to additionally call hasValidShape on the sample environment shapes which are checked in the loop starting on L206